### PR TITLE
[GFX-2194] Typeless texture allocation format support for Depth stencil import 

### PR DESCRIPTION
**Pull request by arx97**
## Jira ticket URL (Why?)
[GFX-2194](https://shapr3d.atlassian.net/browse/GFX-2194)

## Short description (What? How?) 📖
Content's of this PR:

- ANGLE's supported format list extended with DXGI_FORMAT_R32_FLOAT_G8X24_UINT format. I applied a small fix to the generator code, which should handle the TYPELESS textures for our use-case.
- GL_EXT_EGL_image_storage extension support enabled to import textures. ANGLE already supported it based on this: https://chromium.googlesource.com/angle/angle/+/HEAD/doc/ExtensionSupport.md, but they forgot to enable in code.

### Design review 🎨
n/a

### Affected areas 🧭
Angle extension list 
EGL image storage

### Special use-cases to test 🧷
n/a

### How did you test it? 🤔
Manual 💁‍♂️
I checked with the structure pass based selection outline in Shapr3D's Visualization mode.

Automated 💻
n/a

### CODE REVIEW

File: src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ egl::Error ExternalImageSiblingImpl11::initialize(const egl::Display *display)
     mIsRenderable = (textureDesc.BindFlags & D3D11_BIND_RENDER_TARGET) &&
                     (resourceUsage & DXGI_USAGE_RENDER_TARGET_OUTPUT) &&
                     !(resourceUsage & DXGI_USAGE_READ_ONLY);
-
+    mIsDepthAttachment = (textureDesc.BindFlags & D3D11_BIND_DEPTH_STENCIL);
```
Comment (veghbalazs) [inline src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp]: nit: Wdyt of `mIsColorRenderable` and `mIsDepthRenderable`? (Which is pretty much in line with OpenGL terminology).

File: src/libANGLE/renderer/gen_angle_format_table.py [hunk 2]
```diff
@@ -239,6 +239,8 @@ def get_component_alignment_mask(channels, bits):
 
 def get_vertex_attrib_type(format_id):
 
+    if "R32G8X24_TYPELESS" in format_id:
+        return "Float"
```
Comment (veghbalazs) [inline src/libANGLE/renderer/gen_angle_format_table.py]: Can't we let this run into `"InvalidEnum"`? Also curious why is this a `"Float"`, none of the D32S8 formats are allowed for vertex input - in D3D11 at least, do you have a reference for OpenGL?

File: src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp [hunk 3]
```diff
@@ -160,52 +202,55 @@ angle::Result ExternalImageSiblingImpl11::createRenderTarget(const gl::Context *
...
         ANGLE_TRY(mRenderer->allocateResource(context11, rtvDesc, mTexture.get(), &rtv));
         rtv.setInternalName("getAttachmentRenderTarget.RTV");
+        
+        mRenderTarget = std::make_unique<TextureRenderTarget11>(
+            std::move(rtv), mTexture, std::move(srv), std::move(blitSrv), mFormat.info->internalFormat,
+            formatInfo, mWidth, mHeight, 1, mSamples);
     }
-
-    d3d11::SharedSRV srv;
-    if (mIsTexturable)
```
Comment (veghbalazs) [inline src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp:166]: nit: keeping SRV creation for last is likely to yield a less conflict-prone diff

File: src/libANGLE/renderer/gen_angle_format_table.py [hunk 4]
```diff
@@ -345,7 +347,8 @@ def json_to_table_data(format_id, json, angle_to_gl):
     # when adding support for YUV formats that have different identifying markers.
     parsed["isYUV"] = bool_str("PLANE" in format_id)
 
-    parsed["vertexAttribType"] = "gl::VertexAttribType::" + get_vertex_attrib_type(format_id)
+    parsed["vertexAttribType"] = "gl::VertexAttribType::" + \
+        get_vertex_attrib_type(format_id)
```
Comment (palver123) [inline src/libANGLE/renderer/gen_angle_format_table.py]: nit: is there a formatting rule in ANGLE that they broke? Or why did you change the code formatting?

File: src/libANGLE/renderer/Format_table_autogen.cpp [hunk 5]
```diff
@@ -18,7 +18,7 @@ namespace angle
 {
 
 static constexpr rx::FastCopyFunctionMap::Entry BGRAEntry  = {angle::FormatID::R8G8B8A8_UNORM,
-                                                             CopyBGRA8ToRGBA8};
+                                                              CopyBGRA8ToRGBA8};
```
Comment (palver123) [inline src/libANGLE/renderer/Format_table_autogen.cpp]: Where do these formatting changes come from? Visual Studio Code? I think it is better to keep the diff as small as possible and revert these

Comment (veghbalazs) [reply src/libANGLE/renderer/Format_table_autogen.cpp]: These autogen files are auto-formatted by the script, previous committed state didn't adhere to it probably accidentally.

File: src/libANGLE/renderer/gen_angle_format_table.py (hunk 2 above)
Comment (arx97) [reply src/libANGLE/renderer/gen_angle_format_table.py]: I deleted this piece of code to handle `R32G8X24_TYPELESS` similar as other `_TYPELESS` formats: It seems the last member of the generated structure describes the underlying type that determines the alignment, which is UnsignedInt in case of R32 and Byte in case of R8 etc...

(The code still works after the changes)

File: src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp (hunk 1 above)
Comment (arx97) [reply src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp]: https://github.com/shapr3d/angle/pull/20/commits/528ded27ea0852e405efd96cadf55c8a3ba956a4

File: src/libANGLE/renderer/Format_table_autogen.cpp (hunk 5 above)
Comment (arx97) [reply src/libANGLE/renderer/Format_table_autogen.cpp]: It's somehow done by the generator, but I reverted in: https://github.com/shapr3d/angle/pull/20/commits/0076be8a45a6a92300452e8580f33f074fdd53a3

File: src/libANGLE/renderer/gen_angle_format_table.py (hunk 4 above)
Comment (arx97) [reply src/libANGLE/renderer/gen_angle_format_table.py]: VS Code auto-formatted: https://github.com/shapr3d/angle/pull/20/commits/0076be8a45a6a92300452e8580f33f074fdd53a3

File: src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp (hunk 3 above)
Comment (arx97) [reply src/libANGLE/renderer/d3d/d3d11/ExternalImageSiblingImpl11.cpp:166]: When we talked about this I couldn't remember why I swapped the order, but since then I figured it out: The `SRV` is needed for both `RenderTarget` constructor call and I moved these inside of their corresponding if, therefore the `SRV` creation must precede those.

File: src/libANGLE/renderer/gen_angle_format_table.py (hunk 2 above)
Comment (veghbalazs) [reply src/libANGLE/renderer/gen_angle_format_table.py]: As discussed: all typeless should probably be `InvalidEnum` but we're fine with not fixing it (we aim for a smaller diff), since it has no runtime consequence for those types anyway.

**Review by veghbalazs — APPROVED** [review]

**Review by palver123 — APPROVED** [review]
